src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1010,27 +1011,31 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		if (items.length === 0) {
 			return;
 		}
+		const fuzzySearchToggle = new Toggle({
+			title: 'Fuzzy search',
+			icon: Codicon.searchFuzzy,
+			isChecked: filterMode === 'fuzzy',
+			inputActiveOptionBorder: this._themeService.getColorTheme().getColor(inputActiveOptionBorder),
+			inputActiveOptionForeground: this._themeService.getColorTheme().getColor(inputActiveOptionForeground),
+			inputActiveOptionBackground: this._themeService.getColorTheme().getColor(inputActiveOptionBackground)
+		});
+		fuzzySearchToggle.onChange(() => {
+			if (fuzzySearchToggle.checked) {
+				quickPick.hide();
+				this.runRecent(type, 'fuzzy', quickPick.value);
+			} else {
+				quickPick.hide();
```

Comment:
Is this needed? Doesn't this.runRecent show the quick pick again?

Reply:
Maybe not, I'll try